src/repositories/TaskRepository/tasks.repository.ts

```diff
@@ -41,4 +41,17 @@ const findOneTaskByKey = async (
 const findOneTaskById = async (id: string): Promise<Task | null> =>
     await prisma.task.findUnique({ where: { id } });
 
-export { createOneTask, findOneTaskByKey, findOneTaskById };
+const updateOneTask = async (
+    taskUpdateInput: UpdateTaskInput
+): Promise<Task> => {
```

Comment:
Est-ce que ça ne peux pas aussi retourner null si la task n'existe pas ? Simple question

Reply:
Si la task n'existe pas (selon un id) non l'update de prisma ne retourne pas null dans ce cas, une erreur est throw.

On la capture pas à ce niveau. Il y a une gestion des cas d'erreurs en amonts avec notre système de gestion d'exception.

Pertinent de catch les erreurs à ce niveau ? pour les retourner je ne pense pas ?

Reply:
Ah oui ok, dans ce cas il y a des tests a essayer.
Là dans ton service tu fait un call sur DB pour check si task exists, un autre call pour check si taskStatus existe, et un autre pour update, 3 calls pour une opération.
Il faudrait voir si lors de l'update sur des tasks ou taskStatus qui n'existe pas on reçoit une erreur de prisma suffisament descriptive pour qu'on puisse l'utiliser avec nos propres exceptions.
Si prisma dit un truc du genre "task not found" ou "taskStatus not found" c'est ok, on peut faire un check la dessus et throw notre exception en fonction, en revanche si prisma throw une erreur du genre "something went wrong, item not found" ce n'est pas assez clair pour qu'on puisse en faire quelque chose, dans ce cas ta solution actuelle est la plus pertinente.

Pas sur d'avoir été trés clair ici :D

Reply:
Si très clair, j'avais déjà jeté un oeil déjà sur le retour d'erreur. Il y a les codes d'erreur prisma mais est ce qu'il sont liés à cette entité ou bien à un type d'erreur absence d'une ressource (rien de sûr) pour la task et pour la task status (bien que le message parle de task_status_key) je pense de mon point de vue que la verification (métier) au niveau du service rempli déjà bien la tâche. A voir si besoin se fait réellement de réécrire cette partie pour que la couche repo throw les exceptions. Je laisse comme ça pour le moment mais on garde en tête que ça peut faire partie d'un futur change.